dspace-api/src/main/java/org/dspace/discovery/SolrServiceResourceRestrictionPlugin.java

```diff
@@ -159,6 +159,17 @@ public void additionalSearchParameters(Context context, DiscoverQuery discoveryQ
                     resourceQuery.append(" OR g").append(group.getID());
                 }
 
+                // DS-8182: Also allow submitter to be able to search their own submission without explicit READ
+                // permission when they are in their workspace
+                // (check if we need this specific limitation, remove if not needed)
+                boolean isWorkspace = StringUtils.startsWith(
+                        discoveryQuery.getDiscoveryConfigurationName(),
+                        SolrServiceWorkspaceWorkflowRestrictionPlugin.DISCOVER_WORKSPACE_CONFIGURATION_NAME
+                );
+                if (isWorkspace) {
+                    resourceQuery.append(" OR submitter_authority:(" + currentUser.getID() + ")");
+                }
+
```

Comment:
this should be not necessary as workspaceitem and workflowitem keep the appropriate resourcepolicy

Reply:
I agree. But I fail to retrieve the `SolrDocument` of the submission in the search results for workspace because the submitter has no `READ` permission; this makes sure the doc shows up in workspace.